src/filter/http_context.rs

```diff
@@ -63,227 +45,167 @@ impl Filter {
         &self,
         rlp: &RateLimitPolicy,
     ) -> protobuf::RepeatedField<RateLimitDescriptor> {
-        //::protobuf::RepeatedField::default()
-        rlp.gateway_actions
+        rlp.rules
             .iter()
-            .filter(|ga| self.filter_configurations_by_rules(&ga.rules))
-            // flatten the vec<vec<Configurations> to vec<Configuration>
-            .flat_map(|ga| &ga.configurations)
-            // All actions cannot be flatten! each vec of actions defines one potential descriptor
-            .flat_map(|configuration| self.build_descriptor(configuration))
+            .filter(|rule: &&Rule| self.filter_rule_by_conditions(&rule.conditions))
+            // Mapping 1 Rule -> 1 Descriptor
+            // Filter out empty descriptors
+            .filter_map(|rule| self.build_single_descriptor(&rule.data))
             .collect()
     }
 
-    fn filter_configurations_by_rules(&self, rules: &[Rule]) -> bool {
-        if rules.is_empty() {
-            // no rules is equivalent to matching all the requests.
+    fn filter_rule_by_conditions(&self, conditions: &[Condition]) -> bool {
+        if conditions.is_empty() {
+            // no conditions is equivalent to matching all the requests.
             return true;
         }
 
-        rules.iter().any(|rule| self.rule_applies(rule))
+        conditions
+            .iter()
+            .any(|condition| self.condition_applies(condition))
     }
 
-    fn rule_applies(&self, rule: &Rule) -> bool {
-        if !rule.paths.is_empty()
-            && !rule
-                .paths
-                .iter()
-                .any(|path| path_match(path, self.request_path().as_str()))
-        {
-            return false;
-        }
-
-        if !rule.methods.is_empty()
-            && !rule
-                .methods
-                .iter()
-                .any(|method| self.request_method().eq(method))
-        {
-            return false;
-        }
+    fn condition_applies(&self, condition: &Condition) -> bool {
+        condition
+            .all_of
+            .iter()
+            .all(|pattern_expression| self.pattern_expression_applies(pattern_expression))
+    }
 
-        if !rule.hosts.is_empty()
-            && !rule
-                .hosts
-                .iter()
-                .any(|subdomain| subdomain_match(subdomain, self.request_authority().as_str()))
-        {
-            return false;
+    fn pattern_expression_applies(&self, p_e: &PatternExpression) -> bool {
+        let attribute_path = tokenize_with_escaping(&p_e.selector, '.', '\\');
+        // convert a Vec<String> to Vec<&str>
+        // attribute_path.iter().map(AsRef::as_ref).collect()
+        match self.get_property(attribute_path.iter().map(AsRef::as_ref).collect()) {
+            None => {
+                debug!(
+                    "[context_id: {}]: selector not found: {}",
+                    self.context_id, p_e.selector
+                );
+                false
```

Comment:
Not sure if we should default to "pattern expression does NOT apply" (and therefore nor the limit) when it fails to get the property. I see where this can be a problem such as in cases where the operator is `neq`. E.g.:

```yaml
- conditions:
  - allOf:
    - selector: missing.property
       operator: neq
       value: "x"
```

Technically, `missing.property != "x"`.

Reply:
Thanks @alexsnaps!